src/transformers/models/iris/modeling_iris.py

```diff
@@ -0,0 +1,1964 @@
+        self.lpips = LPIPS().eval() if with_lpips else None
+
+    def __repr__(self) -> str:
+        return "tokenizer"
+
+    def forward(
+        self,
+        x: torch.Tensor,
+        output_hidden_states: bool = False,
+        output_attentions: bool = False,
+        should_preprocess: bool = False,
+        should_postprocess: bool = False,
+    ) -> Tuple[torch.Tensor]:
+        outputs, all_hidden_states_enc, attentions_enc = self.encode(
+            x, should_preprocess, output_hidden_states, output_attentions
+        )
+        decoder_input = outputs.z + (outputs.z_quantized - outputs.z).detach()
+        reconstructions, all_hidden_states_dec, attentions_dec = self.decode(
+            decoder_input, should_postprocess, output_hidden_states, output_attentions
+        )
+        return (
+            (outputs.z, outputs.z_quantized, reconstructions),
+            all_hidden_states_enc + all_hidden_states_dec if output_hidden_states else None,
+            attentions_enc + attentions_dec if output_attentions else None,
+        )
+
+    def compute_loss(
+        self, batch: Batch, output_hidden_states: bool = False, output_attentions: bool = False, **kwargs: Any
+    ) -> LossWithIntermediateLosses:
+        assert self.lpips is not None
+
+        observations = self.preprocess_input(torch.flatten(batch["observations"], end_dim=1).contiguous())
+        outputs, all_hidden_states, all_attentions = self(
+            observations, output_hidden_states, output_attentions, should_preprocess=False, should_postprocess=False
+        )
+
+        # Codebook loss. Notes:
+        # - beta position is different from taming and identical to original VQVAE paper
+        # - VQVAE uses 0.25 by default
+        beta = 1.0
+        commitment_loss = (outputs[0].detach() - outputs[1]).pow(2).mean() + beta * (
+            outputs[0] - outputs[1].detach()
+        ).pow(2).mean()
+
+        reconstruction_loss = torch.abs(observations - outputs[2]).mean()
+        perceptual_loss = torch.mean(self.lpips(observations, outputs[2]))
+
+        return (
+            LossWithIntermediateLosses(
+                commitment_loss=commitment_loss,
+                reconstruction_loss=reconstruction_loss,
+                perceptual_loss=perceptual_loss,
+            ),
+            outputs,
+            all_hidden_states,
+            all_attentions,
+        )
+
+    def encode(
+        self,
+        x: torch.Tensor,
+        should_preprocess: bool = False,
+        output_hidden_states: bool = False,
+        output_attentions: bool = False,
+    ) -> TokenizerEncoderOutput:
+        if should_preprocess:
+            x = self.preprocess_input(x)
+        shape = x.shape  # (..., C, H, W)
+        x = x.view(-1, *shape[-3:])
+        z, all_hidden_states, attentions = self.encoder(x, output_hidden_states, output_attentions)
+        z = self.pre_quant_conv(z)
+        b, e, h, w = z.shape
```

Comment:
No single letter var names - these should all be explicit e.g. `batch_size` instead of `b`

Reply:
:+1: sure